src/dynamic_action.rs

```diff
@@ -0,0 +1,142 @@
+//! The dynamic action type.
+//!
+//! This type is primarily intended for use when declaring every single action in a single enum isn't feasible,
+//! for example due to a complex crate hierarchy where various feature crates want to individually declare their own
+//! actions without needing to coordinate with other crates.
+//!
+//! Example:
+//! ```
+//! # use leafwing_input_manager::dynamic_action::{DynActionMarker, DynActionRegistry};
+//!
+//! // In crate `feature_one`
+//! #[derive(DynActionMarker)]
+//! struct FeatureOneAction;
+//!
+//! fn register_feature_one_actions(registry: &mut DynActionRegistry) {
```

Comment:
Hmm, I think this may be clearer with an app extension method.

It's hard for beginners to piece together exactly how you're supposed to store and mutate the `DynActionRegistry`.